pkg/datasource/accessors.go

```diff
@@ -322,186 +334,203 @@ func (a *fieldAccessor) Annotations() map[string]string {
 	return maps.Clone(a.f.Annotations)
 }
 
-func (a *fieldAccessor) Uint8(data Data) uint8 {
+func (a *fieldAccessor) Uint8(data Data) (uint8, error) {
 	val := a.Get(data)
 	if len(val) < 1 {
-		return 0
+		return 0, ErrFieldTooShort
 	}
-	return val[0]
+	return val[0], nil
 }
 
-func (a *fieldAccessor) Uint16(data Data) uint16 {
+func (a *fieldAccessor) Uint16(data Data) (uint16, error) {
 	val := a.Get(data)
 	if len(val) < 2 {
-		return 0
+		return 0, ErrFieldTooShort
 	}
-	return a.ds.byteOrder.Uint16(val)
+	return a.ds.byteOrder.Uint16(val), nil
 }
 
-func (a *fieldAccessor) Uint32(data Data) uint32 {
+func (a *fieldAccessor) Uint32(data Data) (uint32, error) {
 	val := a.Get(data)
 	if len(val) < 4 {
-		return 0
+		return 0, ErrFieldTooShort
 	}
-	return a.ds.byteOrder.Uint32(val)
+	return a.ds.byteOrder.Uint32(val), nil
 }
 
-func (a *fieldAccessor) Uint64(data Data) uint64 {
+func (a *fieldAccessor) Uint64(data Data) (uint64, error) {
 	val := a.Get(data)
 	if len(val) < 8 {
-		return 0
+		return 0, ErrFieldTooShort
 	}
-	return a.ds.byteOrder.Uint64(val)
+	return a.ds.byteOrder.Uint64(val), nil
 }
 
-func (a *fieldAccessor) Int8(data Data) int8 {
+func (a *fieldAccessor) Int8(data Data) (int8, error) {
 	val := a.Get(data)
 	if len(val) < 1 {
-		return 0
+		return 0, ErrFieldTooShort
 	}
-	return int8(val[0])
+	return int8(val[0]), nil
 }
 
-func (a *fieldAccessor) Int16(data Data) int16 {
+func (a *fieldAccessor) Int16(data Data) (int16, error) {
 	val := a.Get(data)
 	if len(val) < 2 {
-		return 0
+		return 0, ErrFieldTooShort
 	}
-	return int16(a.ds.byteOrder.Uint16(val))
+	return int16(a.ds.byteOrder.Uint16(val)), nil
 }
 
-func (a *fieldAccessor) Int32(data Data) int32 {
+func (a *fieldAccessor) Int32(data Data) (int32, error) {
 	val := a.Get(data)
 	if len(val) < 4 {
-		return 0
+		return 0, ErrFieldTooShort
 	}
-	return int32(a.ds.byteOrder.Uint32(val))
+	return int32(a.ds.byteOrder.Uint32(val)), nil
 }
 
-func (a *fieldAccessor) Int64(data Data) int64 {
+func (a *fieldAccessor) Int64(data Data) (int64, error) {
 	val := a.Get(data)
 	if len(val) < 8 {
-		return 0
+		return 0, ErrFieldTooShort
 	}
-	return int64(a.ds.byteOrder.Uint64(val))
+	return int64(a.ds.byteOrder.Uint64(val)), nil
 }
 
-func (a *fieldAccessor) Float32(data Data) float32 {
-	return math.Float32frombits(a.Uint32(data))
+func (a *fieldAccessor) Float32(data Data) (float32, error) {
+	i, err := a.Uint32(data)
+	if err != nil {
+		return 0.0, err
+	}
+	return math.Float32frombits(i), nil
 }
 
-func (a *fieldAccessor) Float64(data Data) float64 {
-	return math.Float64frombits(a.Uint64(data))
+func (a *fieldAccessor) Float64(data Data) (float64, error) {
+	i, err := a.Uint64(data)
+	if err != nil {
+		return 0.0, err
+	}
+	return math.Float64frombits(i), nil
 }
 
-func (a *fieldAccessor) String(data Data) string {
+func (a *fieldAccessor) String(data Data) (string, error) {
```

Comment:
Can this (and `.Bytes()`) ever fail? But yeah, keeping this consistent is probably better than making exceptions.

Reply:
So far they'll never fail, but I think having them consistent is better.